packages/jaeger-ui/src/components/TracePage/CriticalPath/utils/findLastFinishingChildSpanId.tsx

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2023 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Span, Trace } from '../../../../types/trace';
+
+// This function finds LFC of a current span which is less than the spawn event time
+// |-------------spanA--------------|
+//    |--spanB--|    |--spanC--|
+// Here span-C will be LFC of spanA. But spanA doesnt have LFC.So, it returns to parent
+// from its creationTime (startTime) and this is referred as spawnTime below
```

Comment:
Based on the function logic of `findLastFinishingChildSpanId`, we are trying to find the last child which starts and finishes before/equal to the spawnTime?

So based on the example that I drew, for -
* `spawnTime=t1`, spanB starts and finished before time `t1`, so that's the LFC
* `spawnTime=t2`, spanC starts and finished before time `t2`, so that's the LFC
* `spawnTime=t3`, no span starts/finishes before time `t3`, so we have no LFC?

If the above example demonstrates the logic that you've written, then let's use this example to demonstrate which spawnTime are you referring in the example. It's difficult to deduce which spawnTime you want to intend to use.

Also, why do we want to keep spawnTime to be undefined? Can you share an example for this?
```

 spawnTime=t3       spawnTime=t1                       spawnTime=t2
      :                   :                             :
      :                   :                             :
      :                   :                             :
      :                   :                             :
      :                   :                             :
      :                   :                             :
     │:      Span A       :                             : │
     │:───┬───────────▲───┬───────▲────┬─────────────▲──:─┤
     │:   │           │   │       │    │             │  : │
      :   │           │   │       │    │             │  :
      :   │           │   │       │    │             │  :
      :   ▼───────────┘   ▼───────┘    ▼─────────────┘  :
            Span B         Span D           Span C

```

Reply:
Here I am referring spawnTime only when there is no LFC to a span then it focus shifts to its parent span from the span's startTime. This startTime I am referring as spawnTime. 
If there is a LFC to a span then spawnTime is undefined as it is a **optional parameter.**
Example:



     t1                           t4
     ┌─────────────────────────────┐
     │         Span A              │
     └─┬───────▲──────┬─────────▲──┘
       │       │      │         │
      ┌▼───────┴┐  t2┌▼─────────┴┐t3
      │ Span B  │    │ Span C    │
      └─────────┘    └───────────┘


Here initially spanA section from t3 to t4 is added to criticalPathSections as spanC is LFC to span A and here 
``spawn time is undefined`` as there exists a LFC span C to spanA. 
Next we add spanC from t2 to t3 to criticalPathSections. But for spanC there is no LFC our focus shifts to spanA from the time t2.So here ``spawnTime is t2`` and we check any LFC for spanA less than this spawnTime t2

Reply:
Makes sense, can you write up the same in the code documentation? It would help us to understand the business logic you're trying to have.

Reply:
Updated Comments. Let me Know if there are any changes